libraries/exoplayer_rtsp/src/main/java/androidx/media3/exoplayer/rtsp/reader/RtpMp4aReader.java

```diff
@@ -94,23 +94,30 @@ public void consume(
     if(fragmentedSampleSizeBytes > 0 && expectedSequenceNumber < sequenceNumber) {
       outputSampleMetadataForFragmentedPackets();
     }
-    int sampleOffset = 0;
-    for (int subFrame = 0; subFrame <= numSubFrames; subFrame++) {
+    int audioPayloadOffset = 0;
+    for (int subFrame = 0; subFrame < numberOfSubframes; subFrame++) {
       int sampleLength = 0;
 
-      /* Each subframe starts with a variable length encoding */
-      for (; sampleOffset < data.bytesLeft(); sampleOffset++) {
-        sampleLength += data.getData()[sampleOffset] & 0xff;
-        if ((data.getData()[sampleOffset] & 0xff) != 0xff) {
+      /**
+       * This implements PayloadLengthInfo() in Chapter 1.7.3.1, it's only support one program and
+       * one layer.
+       * Each subframe starts with a variable length encoding.
+       */
+      for (; audioPayloadOffset < data.bytesLeft(); audioPayloadOffset++) {
+        int payloadMuxLength = data.peekUnsignedByte();
```

Comment:
my bad, you can actually use `.readUnsignedByte()` to save the increment on L112 (and maybe also 115)